src/probnum/problems/_problems.py

```diff
@@ -1 +1,157 @@
-"""Definitions of problems solved by probabilistic numerical methods."""
+"""Definitions of problems currently solved by probabilistic numerical methods.
+
+If a problem is not listed here, we have not implemented a probabilistic
+numerical method for it yet.
+"""
+
+import dataclasses
+import typing
+
+import numpy as np
+import scipy.sparse
+
+import probnum
+
+
+@dataclasses.dataclass
+class RegressionProblem:
+    r"""Regression problem.
+
+    Fit a stochastic process to data, given a likelihood (realised by a :obj:`Transition`).
+    Solved by Kalman filtering and smoothing in :mod:`probnum.filtsmooth`.
+
+    Parameters
+    ----------
+    observations
+        Observations of the latent process.
+    locations
+        Grid-points on which the observations were taken.
+    likelihood
+        Likelihood of the observations; that is, relation between the latent process and the observed values. Encodes for example noise.
+
+    Examples
+    --------
+    >>> obs = [11.4123, -15.5123]
+    >>> loc = [0.1, 0.2]
+    >>> rp = RegressionProblem(observations=obs, locations=loc)
+    >>> rp
+    RegressionProblem(observations=[11.4123, -15.5123], locations=[0.1, 0.2], likelihood=None)
+    >>> rp.observations
+    [11.4123, -15.5123]
+    """
+
+    observations: np.ndarray
+    locations: np.ndarray
+    likelihood: "probnum.filtsmooth.statespace.DiscreteGaussian" = None
+
+
+@dataclasses.dataclass
+class IVProblem:
+    r"""First order ODE initial value problem.
+
+    Compute a function :math:`y=y(t)` that solves
+
+    .. math::
+        \dot y(t) = f(t, y(t)), \quad y(t_0) = y_0
+
+    on time-interval :math:`[t_0, t_\text{max}]`.
+    Solved by the ODE solvers in :mod:`probnum.diffeq`.
+
+
+    Parameters
+    ----------
+    f
+        ODE vector-field.
+    t0
+        Initial point in time.
+    tmax
+        Final point in time.
+    y0
+        Initial value of the solution.
+
+    Examples
+    --------
+    >>> def f(t, x):
+    ...     return x*(1-x)
+    >>> ivp = IVProblem(f, t0=0., tmax=3., y0=0.1)
+    >>> ivp.t0, ivp.tmax, ivp.y0
+    (0.0, 3.0, 0.1)
+    >>> np.round(ivp.f(ivp.t0, ivp.y0), 2)
+    0.09
+    """
+
+    f: typing.Callable[[float, np.ndarray], np.ndarray]
+    t0: float
+    tmax: float  # Bold move: remove this? This is not really part of an IVP.
```

Comment:
@nathanaelbosch usually tmax is in such definitions. Though, isn't it more elegant to not have it in here? What do you think?

Reply:
Oh btw, this will replace the `IVP` object soon. It does the same, but is so much less code.

Reply:
I don't really see why tmax should not be in there, it seems like a strict part of the problem definition for me. Though I could imagine a `timespan` argument, which is a tuple `(t0, tmax)` to make the function signature a bit compacter. The current api is completely fine too though.

Reply:
I tend to think that only the initial time point is relevant for the problem definition. Only as soon as one starts to think about solutions, the final time point comes into play (existence of solutions, etc.). I see that removing it would be quite an exotic choice, though, so I think putting it in is fine. 

> Though I could imagine a timespan argument, which is a tuple (t0, tmax) to make the function signature a bit compacter. The current api is completely fine too though.

I could add a property that returns this? Or not do it for now, I mean, it is not a gamechanger is it

Reply:
I don't think a property for this is really useful, if anything I'd just prefer it for the creation of the problem. It just feels a bit more balanced, since that way each "component' of the problem gets 1 slot (vector field, initial value, time domain), but again it's quite a minor thing so I think the current implementation is also fine :)

Reply:
In this case I would leave it as is: I believe calling `ivp.t0, ivp.tmax` is more descriptive than `ivp.timespan[0], ivp.timespan[1]`. If you disagree, voice your opinion, but it seems to me that you dont have a super strong opinion either :)